src/ring_theory/integrally_closed.lean

```diff
@@ -137,3 +161,41 @@ begin
 end
 
 end integral_closure
+
+namespace is_integrally_closed
+
+open integral_closure
+
+variables {R : Type*} [comm_ring R] [is_domain R]
+variables (K : Type*) [field K] [algebra R K] [is_fraction_ring R K]
+
+theorem eq_map_of_dvd_of_monic [is_integrally_closed R] {f : R[X]} (hf : f.monic)
+  {g : K[X]} (hg : g.monic) (hd : g ∣ f.map (algebra_map R K)) :
+  ∃ g' : R[X], g'.map (algebra_map R K) = g :=
+begin
+  let algeq := (subalgebra.equiv_of_eq _ _ $
+    is_integrally_closed.integral_closure_eq_bot R _).trans
+    (algebra.bot_equiv_of_injective $ is_fraction_ring.injective R $ K),
+  have : (algebra_map R _).comp algeq.to_alg_hom.to_ring_hom =
+    (integral_closure R _).to_subring.subtype,
+  { ext, conv_rhs { rw ← algeq.symm_apply_apply x }, refl },
+  refine ⟨map algeq.to_alg_hom.to_ring_hom _, _⟩,
+  use g.to_subring _ (frange_subset_integral_closure K hf hg hd),
+  rw [map_map, this],
+  apply g.map_to_subring,
+end
+
+lemma eq_map_mul_C_of_dvd [is_integrally_closed R] {f : R[X]} (hf : f.monic)
```

Comment:
I am not sure, but maybe `eq_map_of_dvd_of_monic` can be stated like this (meaning the existence of a polynomial such that the multiplication blah blah). Since you already used it you know better than me which one is the most useful version.